researchtikpy/query_lang.py

```diff
@@ -0,0 +1,335 @@
+    KG = "KG"
+    PE = "PE"
+    CH = "CH"
+    LV = "LV"
+    PR = "PR"
+    NZ = "NZ"
+    TL = "TL"
+    BT = "BT"
+    MN = "MN"
+    FJ = "FJ"
+    SZ = "SZ"
+    VU = "VU"
+    BF = "BF"
+    TJ = "TJ"
+    BA = "BA"
+    AM = "AM"
+    TD = "TD"
+    SI = "SI"
+    CY = "CY"
+    MW = "MW"
+    EE = "EE"
+    XK = "XK"
+    ME = "ME"
+    KY = "KY"
+    YE = "YE"
+    LS = "LS"
+    ZW = "ZW"
+    MC = "MC"
+    GN = "GN"
+    BS = "BS"
+    PF = "PF"
+    NA = "NA"
+    VI = "VI"
+    BB = "BB"
+    BZ = "BZ"
+    CW = "CW"
+    PS = "PS"
+    FM = "FM"
+    PG = "PG"
+    BI = "BI"
+    AD = "AD"
+    TV = "TV"
+    GL = "GL"
+    KM = "KM"
+    AW = "AW"
+    TC = "TC"
+    CV = "CV"
+    MO = "MO"
+    VC = "VC"
+    NE = "NE"
+    WS = "WS"
+    MP = "MP"
+    DJ = "DJ"
+    RW = "RW"
+    AG = "AG"
+    GI = "GI"
+    GQ = "GQ"
+    AS = "AS"
+    AX = "AX"
+    TO = "TO"
+    KN = "KN"
+    LC = "LC"
+    NC = "NC"
+    LI = "LI"
+    SS = "SS"
+    IR = "IR"
+    SY = "SY"
+    IM = "IM"
+    SC = "SC"
+    VG = "VG"
+    SB = "SB"
+    DM = "DM"
+    KI = "KI"
+    UM = "UM"
+    SX = "SX"
+    GD = "GD"
+    MH = "MH"
+    BQ = "BQ"
+    YT = "YT"
+    ST = "ST"
+    CF = "CF"
+    BM = "BM"
+    SM = "SM"
+    PW = "PW"
+    GU = "GU"
+    HK = "HK"
+    IN = "IN"
+    CK = "CK"
+    AQ = "AQ"
+    WF = "WF"
+    JE = "JE"
+    MQ = "MQ"
+    CN = "CN"
+    GF = "GF"
+    MS = "MS"
+    GG = "GG"
+    TK = "TK"
+    FK = "FK"
+    PM = "PM"
+    NU = "NU"
+    MF = "MF"
+    ER = "ER"
+    NF = "NF"
+    VA = "VA"
+    IO = "IO"
+    SH = "SH"
+    BL = "BL"
+    CU = "CU"
+    NR = "NR"
+    TP = "TP"
+    BV = "BV"
+    EH = "EH"
+    PN = "PN"
+    TF = "TF"
+    RU = "RU"
+
+
+class Fields(StrEnum):
+    """Fields to query"""
+    create_date = "create_date"
+    username = "username"
+    region_code = "region_code"
+    video_id = "video_id"
+    hashtag_name = "hashtag_name"
+    keyword = "keyword"
+    music_id = "music_id"
+    effect_id = "effect_id"
+    video_length = "video_length"
+
+
+class Operators(StrEnum):
+    """Operators to use in query"""
+    equals = "EQ"
+    isin = "IN"
+    greater = "GT"
+    greater_equals = "GTE"
+    larger = "LT"
+    larger_equals = "LTE"
+
+
+@dataclass
+class Condition:
+    """TikTok Research API Query Condition
+
+    Args
+    ----
+    field_name: Fields
+        Field to query
+    operation: Operators
+        Operator to use
+    field_value: Union[str, int, List[str]]
+        Value to compare against
+    """
+
+    field_name: Fields
+    operation: Operators
+    field_values: List[str]
+
+    def __post_init__(self):
+        if self.operation == "EQ" and len(self.field_values) != 1:
+            raise ValueError("EQ operation must have one field value")
+        if self.operation == "IN" and self.field_name == "username" and len(self.field_values) >= 100:
+            raise ValueError("IN operation on username must have less than equal 100 field values")
```

Comment:
I think we should leave this validation to the TikTok research API, since they could drop constraints or add new at any moment

Reply:
Yeah, you're right. I'll remove the `__post_init__` method.